auth-service/src/main/kotlin/uk/gov/communities/delta/auth/services/UserService.kt

```diff
@@ -0,0 +1,136 @@
+package uk.gov.communities.delta.auth.services
+
+import com.google.common.base.Strings
+import org.slf4j.LoggerFactory
+import uk.gov.communities.delta.auth.config.LDAPConfig
+import uk.gov.communities.delta.auth.plugins.UserVisibleServerError
+import java.io.UnsupportedEncodingException
+import javax.naming.directory.*
+
+class UserService(
+    private val ldapService: LdapService,
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    suspend fun createUser(adUser: ADUser) {
+        try {
+            addUserToAD(adUser)
+        } catch (e: Exception) {
+            logger.error("Error creating user with dn {}", adUser.dn, e)
+            throw e
+        }
+    }
+
+    private suspend fun addUserToAD(adUser: ADUser) {
+        val container: Attributes = BasicAttributes()
+        container.put(adUser.objClasses)
+        container.put(BasicAttribute("userPrincipalName", adUser.userPrincipalName))
+        container.put(BasicAttribute("cn", adUser.cn))
+        container.put(BasicAttribute("sn", adUser.sn))
+        container.put(BasicAttribute("givenName", adUser.givenName))
+        container.put(BasicAttribute("mail", adUser.mail))
+        container.put(BasicAttribute("st", adUser.st))
+        container.put(BasicAttribute("userAccountControl", adUser.userAccountControl))
+        if (adUser.password != null) container.put(ADUser.getPasswordAttribute(adUser.password!!))
+        if (adUser.comment != null) container.put(BasicAttribute("comment", adUser.comment))
+
+        val enabled = adUser.userAccountControl == ADUser.accountFlags(true)
+        if (enabled && adUser.password == null) {
+            logger.error("Trying to create an enabled user without a password")
+            throw UserVisibleServerError(
+                "passwordless_user",
+                "Trying to create user with no password",
+                "Something went wrong",
+                "Registration Error"
+            )
```

Comment:
This should never happen right, it would be a bug? In which case can save a few lines and throw a regular `Exception` without logging anything first